### PR TITLE
 Json and code for edx.librarycontentblock.content.assigned event

### PR DESCRIPTION
**Pull request by tehreem-sadat**
**Trello Link:** _https://trello.com/c/yiKJpr4R/129-content-library-interaction-event-edxlibrarycontentblockcontentassigned_

**Description:** _The server emits an edx.librarycontentblock.content.assigned event the first time that content from a randomized content block is delivered to a user. The edx.librarycontentblock.content.assigned event identifies the components delivered from the library to a user._

**Checks before merge:**

- [ ] Reviewed
- [ ] Commits squashed

### CODE REVIEW

File: openedx/features/caliper_tracking/tests/expected/edx.librarycontentblock.content.assigned.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{
+  "@context": "http://purl.imsglobal.org/ctx/caliper/v1p1",
+  "action": "Submitted",
```
[the remark this answers is not in the document]
Comment (tehreem-sadat) [reply openedx/features/caliper_tracking/tests/expected/edx.librarycontentblock.content.assigned.json]: "Showed" can not be used as an action for this event as it  is called when user just click on a course link to enter in that specific course before going to that specific assessment or problem page.

[the remark this answers is not in the document]
Comment (tehreem-sadat) [reply openedx/features/caliper_tracking/tests/expected/edx.librarycontentblock.content.assigned.json]: Moreover I have used "Submitted" as it's definition says "Hand over formally".

File: openedx/features/caliper_tracking/tests/expected/edx.librarycontentblock.content.assigned.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  "actor": {
+    "id": "http://localhost:18000/",
+    "type": "SoftwareApplication"
+  },
+  "eventTime": "2018-10-18T15:34:59.194Z",
+  "extensions": {
+    "extra_fields": {
+      "accept_language": "en-US,en;q=0.5",
+      "agent": "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:62.0) Gecko/20100101 Firefox/62.0",
+      "event_source": "server",
```
Comment (mumarkhan999) [inline openedx/features/caliper_tracking/tests/expected/edx.librarycontentblock.content.assigned.json]: `course_id` and `course_user_tags` are missing.

Comment (tehreem-sadat) [reply openedx/features/caliper_tracking/tests/expected/edx.librarycontentblock.content.assigned.json]: done

File: openedx/features/caliper_tracking/tests/expected/edx.librarycontentblock.content.assigned.json [hunk 3]
```diff
@@ -0,0 +1,45 @@
+{
+  "@context": "http://purl.imsglobal.org/ctx/caliper/v1p1",
+  "action": "Submitted",
+  "actor": {
```
[the remark this answers is not in the document]
Comment (asamolion) [reply openedx/features/caliper_tracking/tests/expected/edx.librarycontentblock.content.assigned.json]: Like the `org_id` or any other such information that can give us more information about the SoftwareApplication.

File: openedx/features/caliper_tracking/tests/expected/edx.librarycontentblock.content.assigned.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (asamolion) [reply openedx/features/caliper_tracking/tests/expected/edx.librarycontentblock.content.assigned.json]: I see. However, `Submitted` in web terms is more akin to something a user would do, while the actor in this case is `SoftwareApplication`, which is showing something to the user.

File: openedx/features/caliper_tracking/tests/expected/edx.librarycontentblock.content.assigned.json [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      "page": null,
+      "path": "/courses/course-v1:Arbisoft+1+1/course/",
+      "session": "02d42e65f86ea01aa03e657ff5e0f758",
+      "user_id": 14,
+      "username": "testuser"
+    }
+  },
+  "id": "urn:uuid:e5891791-3d27-4df1-a272-091806a43dfb",
+  "object": {
+    "extensions": {
```
Comment (asamolion) [inline openedx/features/caliper_tracking/tests/expected/edx.librarycontentblock.content.assigned.json]: Don't we have to add information about the entity being linked to i.e. user?

File: openedx/features/caliper_tracking/tests/expected/edx.librarycontentblock.content.assigned.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (danialmalik) [reply openedx/features/caliper_tracking/tests/expected/edx.librarycontentblock.content.assigned.json]: Like this ?
```
    "name": "edX"
```

File: openedx/features/caliper_tracking/tests/expected/edx.librarycontentblock.content.assigned.json (hunk 4 above)
Comment (tehreem-sadat) [reply openedx/features/caliper_tracking/tests/expected/edx.librarycontentblock.content.assigned.json]: done.

**Pull request merged**